blaze/compute/sql.py

```diff
@@ -952,6 +953,42 @@ def compute_up(expr, data, **kwargs):
                               group_by=grouper)
 
 
```

Comment:
Can you refactor the three `compute_up()` implementations to call a common internal function, since they all have the same core logic?